pkg/labels/labels.go

```diff
@@ -18,12 +18,21 @@ package labels
 
 // LabelUncompressed is added to compressed layer contents.
 // The value is digest of the uncompressed content.
-const LabelUncompressed = "containerd.io/uncompressed"
+const (
+	LabelUncompressed = "containerd.io/uncompressed"
 
-// LabelSharedNamespace is added to a namespace to allow that namespaces
-// contents to be shared.
-const LabelSharedNamespace = "containerd.io/namespace.shareable"
+	// LabelSharedNamespace is added to a namespace to allow that namespaces
+	// contents to be shared.
+	LabelSharedNamespace = "containerd.io/namespace.shareable"
 
-// LabelDistributionSource is added to content to indicate its origin.
-// e.g., "containerd.io/distribution.source.docker.io=library/redis"
-const LabelDistributionSource = "containerd.io/distribution.source"
+	// LabelDistributionSource is added to content to indicate its origin.
+	// e.g., "containerd.io/distribution.source.docker.io=library/redis"
+	LabelDistributionSource = "containerd.io/distribution.source"
+
+	// PlatformLabelPrefix is common prefix for indicating what platforms
+	// an image has been pulled for
+	PlatformLabelPrefix = "containerd.io/imagePulledForPlatform"
```

Comment:
I'm not sure I follow the design here. Would we just use existing API to check the platform part?
IMO, it's hard to make sure that the existing data is consistent with label. 
The user might use `ctr` or other tools to pull images and change the labels.

Reply:
I think just like how we add CRI image labels for images pulled through ctr, we should do the same for platform label as well. We need some way to track which platform the image is being pulled for as it is needed by ImageService.UpdateImage() that gets called after an event is published. An alternative was to store this information as an exclusive field in containerd image struct but feedback given was that runtimehandler/platform info is specific to CRI and should not be saved to core containerd structures. Therefore, labels are the only alternative to achieve this.

@dmcgowan